lib/middleware/common.js

```diff
@@ -99,22 +99,7 @@ function setHeavyCacheHeaders (response) {
   response.setHeader('Cache-Control', 'public, max-age=31536000')
 }
 
-class PromiseContainer {
-  constructor () {
-    this.promise = null
-  }
-
-  then (success, error) {
-    return this.promise.then(success, error)
-  }
-
-  set (newPromise) {
-    this.promise = newPromise
-  }
-}
-
 // PUBLIC API
-exports.PromiseContainer = PromiseContainer
```

Comment:
I wonder if this could break users who are using this in plugins.

Can you put the delete in a separate PR and comment that it is deprecated in this PR?

Reply:
done